### PR TITLE
Add logistic regression example on iris dataset.

### PR DESCRIPTION
**Pull request by JoshuaWhittemore**
This is a super simple model aimed at helping beginners get started with Flux.

It depends on an 'iris' Data module in Flux, PR here: https://github.com/FluxML/Flux.jl/pull/652.

If it doesn't make sense to have the iris Data module in Flux I can move that over here.

Please make any suggestions/changes you think might be appropriate.  Thank you!

### CODE REVIEW

File: other/iris/iris.jl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+)
+
+loss(x, y) = crossentropy(model(x), y)
+
+# Gradient descent optimiser with learning rate 0.5.
+optimiser = Descent(0.5)
+
+
+# Start Training.
+for epoch in 1:100
```
Comment (MikeInnes) [inline other/iris/iris.jl]: This could be done with something like `data = Iterators.repeat((X, Y), 100)`.

**Comment by joshua-whittemore** [comment]
Hi Mike, I've changed the call to `train!` to use an iterator as you suggested.  Let me know if you have any other suggestions.

One thing - this model depends on FluxML/Flux.jl#652 for its dataset.  Thank you!

**Comment by MikeInnes** [comment]
That PR is merged, so presumably this just needs the manifest to be updated?

**Comment by JoshuaWhittemore** [comment]
Yes I think so.  I was going to do that today.  I just need to figure out how to require that it use Flux 0.8.1+ I think.

**Comment by JoshuaWhittemore** [comment]
Err, 0.8.1 or 0.8.1+ I guess.

**Comment by JoshuaWhittemore** [comment]
I've updated the manifest.  Let me know if there's anything else I should do.  Thanks!

**Comment by MikeInnes** [comment]
Thanks!

**Pull request merged**